subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -424,6 +424,63 @@ static int cmd_wifi_ap_disable(const struct shell *sh, size_t argc,
 	return 0;
 }
 
+
+static int cmd_wifi_reg_domain(const struct shell *sh, size_t argc,
+			       char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_reg_domain regd = {0};
+	int ret;
+
+	if (argc == 1) {
+		regd.oper = WIFI_MGMT_GET;
+	} else if (argc >= 2) {
+		regd.oper = WIFI_MGMT_SET;
+		if (strlen(argv[1]) != 2) {
+			shell_fprintf(sh, SHELL_WARNING, "Invalid reg domain %c%c\n", argv[1][0],
+				      argv[1][1]);
+			return -ENOEXEC;
+		}
+
+		/* Two letter country code with special case of 00 for WORLD */
+		if ((argv[1][0] < 'A' || argv[1][0] > 'Z') ||
+		    (argv[1][1] < 'A' || argv[1][1] > 'Z') ||
+			(argv[1][0] == '0' || argv[1][1] == '0')) {
```

Comment:
This condition appears to be wrong. Any `'0'` character would result in this warning print, also `00`.